### DISCUSSION TITLE
Web search enhancement suggestion (enter non-English keywords → translate into English → search for English keywords)

### DISCUSSION THREAD

**Discussion by photkey** [Flow-Launcher/Flow.Launcher#721] [Ideas]
Add a translation function based on the existing function, similar to https://english.sogou.com/, input non-English keywords, automatically translated into English as search keywords Open the search page.

Application scenario.
Newbies just learning programming, searching for some programming related issues, often need to translate into English first and then search, which is tedious.

Translated with www.DeepL.com/Translator (free version)

网页搜索插件功能增强建议（输入非英文关键词→翻译成英文→搜索英文关键词）

在现有功能基础上增加一个翻译功能，类似https://english.sogou.com/ ，输入非英文关键词，自动翻译成英文作为搜索关键词 打开搜索页面。

应用场景：
新手刚学编程（且英文水平极弱……），搜索一些编程相关的问题，经常需要先翻译成英文后再搜索，很繁琐。中间翻译这一步，如果可以用插件自动完成，可以方便很多。

**Comment by taooceros** [comment]
Why not just add https://english.sogou.com/ as a search source? That even doesn't need to modify flow's code.

**Reply by photkey** [reply]
1, the first is not good to add, "https://english.sogou.com/english?query={q}" this form of access will trigger the anti-crawler mechanism, you need to enter the verification code.
2, Google and GitHub are more likely to find satisfactory answers.

1、首先是不好加，“https://english.sogou.com/english?query={q}” 这种形式访问的话 会触发反爬虫机制，需要输入验证码；
2、Google和GitHub更容易找到满意的答案；

**Reply by taooceros** [reply]
That's true. 
Though, it is not easy to implement the feature because a translator is needed.
Anyway, pull request is welcome.

**Comment by taooceros** [comment]
Just an idea. If you would like to implement this, you can add another search suggestion to translate the query test so no need for extra settings.

https://github.com/Flow-Launcher/Flow.Launcher/tree/dev/Plugins/Flow.Launcher.Plugin.WebSearch/SuggestionSources

**Reply by photkey** [reply]
Thanks, I only have a little bit of introductory python, and after checking the file suffix, this seems to be C# code, which I don't understand at all; but I'm willing to spend some time trying to study it (because this application scenario is really frequent, almost every day, for someone like me with very poor English).
Thank you very much for your reply.

谢谢，我只有一点python入门的基础，查了一下文件后缀，这好像是C#的代码，完全不懂；不过我愿意花点时间 尝试研究一下（因为这个应用场景对我这种英文很差的人，确实是很频繁，几乎是每一天）。
非常感谢你的回复。

**Reply by taooceros** [reply]
如果你更愿意写python的话可以尝试一下写一个python的插件用来实现这个功能
https://flow-launcher.github.io/docs/#/develop-py-plugins?id=how-to-develop-python-plugins
目前文档还不是很齐，所以可能要多研究一下

**Reply by taooceros** [reply]
https://github.com/Flow-Launcher/docs/blob/update_py_docs/develop-py-plugins.md
或者参考这个branch的文档
